proto/p4/p4runtime.proto

```diff
@@ -138,13 +138,24 @@ message TableEntry {
   uint64 controller_metadata = 5;
   // meter_config and counter_data are convenience fields that enable the
   // controller to configure the direct resources associated with the table at
-  // the same time as a match-action entry is inserted or modified. When
-  // updating a table entry, leaving these fields unset means that the direct
-  // resources (if any) will not be updated. When reading a table entry, the
-  // values associated with the direct resources (if any) will be returned. If
-  // the table does not support a direct resource, then setting the
-  // corresponding direct resource field in any table write operation will
-  // return an error.
+  // the same time as a match-action entry is inserted or modified.
+  // Table write:
```

Comment:
This part is LGTM. I suggest also adding comments to clarify important related semantics, i.e. INSERT and DELETE on direct resources is not allowed and will return an error. The creation/deletion (or initializion/reset) should happen as part of INSERT/DELETE on the associated table-entry (I believe this was also part of the consensus).
For clarity, I suggest adding this comment on both DirectMeterEntry as well as DirectCounterEntry messages.

Reply:
Added the comments on the DirectMeterEntry as well as DirectCounterEntry messages as per suggestion. PTAL. Not sure if we should say anything about initialization/reset. The resource is inserted and initialized as part of TableEntry insertion, and we can modify it (including reset) using either TableEntry or  DirectMeterEntry/DirectCounterEntry.